src/main/scala/fr/acinq/bitcoin/Script.scala

```diff
@@ -294,12 +299,16 @@ object Script {
 
   def isPayToScript(script: Array[Byte]): Boolean = script.length == 23 && script(0) == elt2code(OP_HASH160).toByte && script(1) == 0x14 && script(22) == elt2code(OP_EQUAL).toByte
 
-  def removeSignature(script: List[ScriptElt], signature: BinaryData): List[ScriptElt] = {
+  def removeSignature(script: List[ScriptElt], signature: BinaryData): (List[ScriptElt], Boolean) = {
```

Comment:
Is it really necessary ? You can tell is something has been removed by comparing the output and input